app/models/core/ResponseModel.scala

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package models.core
+
+import play.api.Logger
+import play.api.http.Status.{INTERNAL_SERVER_ERROR, NOT_FOUND, OK}
+import play.api.libs.json.{Format, JsSuccess, JsValue}
+import uk.gov.hmrc.http.{HttpReads, HttpResponse}
+
+import scala.util.Try
+
+
+object ResponseModel {
```

Comment:
these models/types specifically related to CreditModel read functionality, if we can add relevant prefix to all of the name, as with the current naming I am under impression that these are generic types we can use everythere

Reply:
These are generic types that can be reused. It's subclassed in CreditsModel:

`implicit val reads: CreditsResponseReads = new CreditsResponseReads

  class CreditsResponseReads extends AResponseReads[CreditsModel] {
    implicit val format: Format[CreditsModel] = CreditsModel.format
  }`

It definitely works with HttpClientV2, haven't tried it with V1 - The Reads for ClaimToAdjust could be refactored to use it pretty simply as that's already on V2. It should hopefully come in useful when we start doing the work to move over to V2 throughout, and cut down on code duplication.